crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -25,18 +26,36 @@ pub struct CoinGecko {
     base_url: Url,
     api_key: Option<String>,
     chain: String,
+    native_price: NativePrice,
+}
+
+enum NativePrice {
```

Comment:
nit: Since we are already in the module context of native prices, I think the technical term for this is [QuoteToken](https://corporatefinanceinstitute.com/resources/economics/currency-pair/)